File: R/functions/genMatrices.R
genMatrices <- function(S) {
  
  # pre-allocate multi-dimensional vectors
  M <- list(gradient = matrix(0, nrow = S$n_items, ncol = S$n_items),
            r = rep(1, S$n_items),
            score = rep(0, S$n_items))
  
  return(M)
  
}

File: MATLAB/functions/genMatrices.m
function M = genMatrices(S)

% here we pre-allocate memory for all multi-dimensional structures we are
% gonna use
M.gradient = zeros(S.n_items, S.n_items);
M.r = ones(1, S.n_items);
M.score = zeros(1, S.n_items);

end